Add a string concatenation to _read_body()

msestak opened the pull request:
I found that this fixes random breakage of results while selecting
from a database.

Fixes issue #2

String concatenation may not be best solution for large selects because
it builds large variable in memory (propose better solution if available).

msestak committed:
Add a string concatenation to _read_body()

I found that this fixes random breakage of results while selecting
from a database.

Fixes issue #2

String concatenation may not be best solution for large selects because
it builds large variable in memory (propose better solution if available).

hulponot commented:
Without this fix, it's impossible to use module with large responses. This changes definitely should be on CPAN

msestak commented:
More elegant solution is offered in this thread:
http://www.perlmonks.org/?node_id=1175544
and I updated this pull request to reflect that.

msestak committed:
Add more elegant solution with remainder

Solution contributed by Corion at http://www.perlmonks.org/?node_id=1175544
This is low memory solution.

Warning added in case that read_entity_body() returns an empty string
but $remainder still has content. In my opinion this shouldn't happen,
but at least we will see it.
